packages/Webkul/Product/src/Repositories/ProductRepository.php

```diff
@@ -71,7 +71,20 @@ public function update(array $data, $id, $attribute = 'id')
     {
         $product = $this->findOrFail($id);
 
+        if (isset($data['variants']) && is_array($data['variants'])) {
+            foreach ($data['variants'] as $key => $variants) {
+                $data['variants'][$key] = array_merge(
+                    $variants,
+                    [
+                        'tax_category_id' => isset($data['tax_category_id']) ? $data['tax_category_id']:null,
+                    ]
+                );
+            }
+        }
+
```

Comment:
Move this code to product's individual instance type,
https://github.com/bagisto/bagisto/blob/master/packages/Webkul/Product/src/Type/Configurable.php#L183